### ISSUE TITLE
LightGBM summary_plot "dot" with binary objective

### ISSUE THREAD

**Issue by aa-tolmachev** [shap/shap#796]
> Here is a simple code - 
> from sklearn.metrics import accuracy_score, roc_auc_score, confusion_matrix
> 
> params = {'metric' : 'auc',
>           'boosting_type' : 'gbdt',
>           'colsample_bytree' : 0.9234,
>           'num_leaves' : 13,
>           'max_depth' : -1,
>           'n_estimators' : 200,
>           'min_child_samples': 399, 
>           'min_child_weight': 0.1,
>           'reg_alpha': 2,
>           'reg_lambda': 5,
>           'subsample': 0.855,
>           'verbose' : -1,
>           'num_threads' : 4,
>           'objective' : 'binary'
> }
> 
> model = lgb.train(params, lgb_data_train, num_boost_round = 600  )
> 
> 
> y_hat = model.predict(df_total[interest_columns])
> score = roc_auc_score(df_total['is_trash'], y_hat)
> print("Overall AUC: {:.3f}" .format(score) )
> 
> 
> shap_values = shap.TreeExplainer(model).shap_values(df_total[interest_columns])
> 
> shap.summary_plot(shap_values, df_total[interest_columns])

target is 0 and 1 like standart LightGBM documentation.
Summary_plot  make a bars graph only (image below)
If we set plot_type = 'dots' then get an error (image below)
<img width="716" alt="Снимок экрана 2019-09-06 в 15 47 04" src="https://user-images.githubusercontent.com/28034900/64422536-bffdfa00-d0bd-11e9-91b8-594bf254b588.png">
<img width="1015" alt="Снимок экрана 2019-09-06 в 15 47 20" src="https://user-images.githubusercontent.com/28034900/64422537-bffdfa00-d0bd-11e9-8eb1-fc3e5cf38214.png">

**Comment by candalfigomoro** [comment]
I have the same issue.

SHAP for LightGBM binary classification is broken in v0.30.x

I think this issue could be related to PR #690 @slundberg @imatiach-msft

**Comment by candalfigomoro** [comment]
The summary plot in https://github.com/slundberg/shap/blob/master/notebooks/tree_explainer/Census%20income%20classification%20with%20LightGBM.ipynb is also wrong, because it displays the bar plot instead of what it is supposed to show.

Maybe we should pass only the shap values for the "1" class.

**Comment by candalfigomoro** [comment]
It looks like specifying the class does the trick (see https://github.com/slundberg/shap/issues/837#issuecomment-539491822), however I don't know if this is consistent with the general API. For scikit-learn's classifiers with KernelExplainer we don't specify the class (see https://slundberg.github.io/shap/notebooks/Census%20income%20classification%20with%20scikit-learn.html), so this should probably be managed internally by TreeExplainer.

**Comment by LucidaScalzi** [comment]
I got the warning 'LightGBM binary classifier with TreeExplainer shap values output has changed to a list of ndarray' after 'shap_values()'. It seems that LGB model has two outputs (one for each class), shap_values returns a list of two arrays. Each array explains each of the outputs of the model. So yes, I think we should pass only the shap values for the "1" class.
